framework/include/actions/ActionWarehouse.h

```diff
@@ -122,20 +122,23 @@ class ActionWarehouse :
     }
 
   /**
-   * Retrieve all actions in a specific type.
+   * Retrieve all actions in a specific type ordered by their names.
    */
   template <class T>
-  std::set<const T *> getActions()
+  std::vector<const T *> getActions()
     {
-      std::set<const T*> actions;
-      for (unsigned int i=0; i<_all_ptrs.size(); i++)
+      std::map<std::string, const T*> actions;
+      for (unsigned int i = 0; i < _all_ptrs.size(); ++i)
       {
         Action * act = _all_ptrs[i].get();
         T* p = dynamic_cast<T*>(act);
         if (p)
-          actions.insert(p);
+          actions.insert(std::pair<std::string, const T*>(act->name(), p));
       }
-      return actions;
+      std::vector<const T*> action_vector;
+      for (auto & pair : actions)
+        action_vector.push_back(pair.second);
+      return action_vector;
```

Comment:
What do you sort? He needs a vector of pointers in the end, so he needs to keep track of the link between the pointer and the name. 

So there will need to be an pointer container and a name container, the names are sorted, then connected back to the pointers. There is probably more logic in that then just using a map.